src/db.c

```diff
@@ -625,13 +626,28 @@ void flushdbCommand(client *c) {
 #endif
 }
 
+static int flushCommandsInternal(client *c) {
+    int flags;
+    if (getFlushCommandFlags(c,&flags) == C_ERR) return C_ERR;
+    flushAllDataAndResetRDB(flags);
+    return C_OK;
+}
+
 /* FLUSHALL [ASYNC]
  *
  * Flushes the whole server data set. */
 void flushallCommand(client *c) {
-    int flags;
-    if (getFlushCommandFlags(c,&flags) == C_ERR) return;
-    flushAllDataAndResetRDB(flags);
+    if (flushCommandsInternal(c) == C_ERR) return; /* Error was already sent to the client */
+    addReply(c,shared.ok);
+}
+
+/* FLUSHEVERYTHING [ASYNC]
+ *
+ * Flushes the whole server data set and functions. */
+void flusheverythingCommand(client *c) {
+    if (flushCommandsInternal(c) == C_ERR) return; /* Error was already sent to the client */
+    /* Clear functions */
+    functionsCtxClear(functionsCtxGetCurrent());
```

Comment:
i suppose we want to flush the EVAL scripts too? in which case maybe we wanna move that into flushAllDataAndResetRDB and control it via some flag?
or alternatively make a dedicated function, and use it also when rdbLoadRio fails?

Reply:
Thanks, pushed a commit to fix and also fixed the top comment.